compiler/rustc_hir_typeck/src/intrinsicck.rs

```diff
@@ -48,8 +48,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let to = normalize(to);
         trace!(?from, ?to);
         if from.has_non_region_infer() || to.has_non_region_infer() {
+            // Note: this path is currently not reached in any test, so any
+            // example that triggers this would be worth minimizing and
+            // converting into a test.
             tcx.dcx().span_delayed_bug(span, "argument to transmute has inference variables");
```

Comment:
should this not be a `span_bug`, after removing the `return` we now use the below code even if there are infer vars